precode.go

```diff
@@ -5,76 +5,98 @@ import (
 	"fmt"
 	"log"
 	"sync"
+	"time"
 )
 
-// Generator генерирует последовательность чисел 1,2,3 и т.д. и
-// отправляет их в канал ch. При этом после записи в канал для каждого числа
-// вызывается функция fn. Она служит для подсчёта количества и суммы
-// сгенерированных чисел.
 func Generator(ctx context.Context, ch chan<- int64, fn func(int64)) {
-	// 1. Функция Generator
-	// ...
+	var i int64 = 1
+	for {
+		select {
+		case <-ctx.Done():
+			close(ch)
+			return
+		default:
+			ch <- i
+			fn(i)
+			i++
+		}
+	}
 }
 
-// Worker читает число из канала in и пишет его в канал out.
-func Worker(in <-chan int64, out chan<- int64) {
-	// 2. Функция Worker
-	// ...
+func Worker(ctx context.Context, in <-chan int64, out chan<- int64) {
```

Comment:
Контекст нужен для прекращения работы генератора, при этом воркер работает пока не закроется канал, так что контекст тут не нужен
```suggestion
func Worker(in <-chan int64, out chan<- int64) {
```